torchbox/torchbox.com
# Fix to reduce duplicate queries on Work and Blog page load

### Description of Changes Made

This PR changes the Work, HistoricWork and Blog Page Models to reduce the number of database queries on page load.

The blog page author is now pre-fetched, and the `HistoricalWorkPage` `feed_image` field has been renamed to header_image to match the image field of the `WorkPage`, allowing the pre-fetch of this common field.

### How to Test

1. Pull the latest staging data if you haven't already done so.
2. Start the site locally and note the number of queries when loading the Work and Blog pages using `django-debug-toolbar` (pre-installed).
3. (Admin task) Ensure at least one HistoryWorkPage has a viewable image set in the feed_image field of the promote panel.
4. (Admin task) Ensure at least one WorkPage has a viewable image set in the feed_image field of the promote panel.
5. Checkout this brach and apply the latest migrations to the Work app
6. Visit the Work and Blog pages again and the number of queries on page load sould be significantally reduced.

### Screenshots

<details>
  <summary>Expand to see more</summary>

#### Before:
![Screenshot from 2024-03-12 17-55-15](https://github.com/torchbox/torchbox.com/assets/52448404/7ad788ac-0049-425e-b45b-a23a87a27a40)



#### After:
![Screenshot from 2024-03-12 17-54-23](https://github.com/torchbox/torchbox.com/assets/52448404/7ebc39d0-b52a-4894-9646-50a22ac05000)


</details>

### MR Checklist


#### Unit tests


#### Documentation


#### Browser testing

  - Latest version of Chrome on Linux

#### Data protection


#### Accessibility


#### Sustainability


#### Pattern library
